### PR TITLE
Datastore repository factory and config settings

### LINKED ISSUE #898: Datastore repository instantiation classes
These are the classes that are somewhat boilerplate implementations of Spring Data interfaces that are involved in creating instances of repositories for the user's types and methods.

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
fixes #898

These are boilerplate type classes that tell Spring the names of the repository, template, and some annotation classes.

### CODE REVIEW

File: spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@Documented
+@Import(DatastoreRepositoriesRegistrar.class)
+public @interface EnableDatastoreRepositories {
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java]: From the past we found out that all of these many default values are needed or instantiation of repositories will fail.

**Comment by meltsufin** [comment]
Can we go over this one in a meeting?

File: spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java (hunk 1 above)
Comment (meltsufin) [inline spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java]: Can we extend `org.springframework.data.repository.config.EnableRepositories`?

Comment (ChengyuanZhao) [reply spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java]: @meltsufin I'm having trouble finding this file, but I did find this: 

https://github.com/spring-projects/spring-data-commons/blob/master/src/test/java/org/springframework/data/repository/config/EnableRepositories.java

Was this the one? This one seems to just be a test file used with the test "DummyRegistrar"

Comment (meltsufin) [reply spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/config/EnableDatastoreRepositories.java]: Oops :)

**Pull request merged**